.github/helpers/jira-api/refine_dependency.py

```diff
@@ -0,0 +1,178 @@
+
+def decode_github_env( encoded_str ):
+    """
+    Used to decode the environment variable that is produced from 
+     .github/helpers/check-npm-dependencies.js
+    See workflow job: check-versions and create-issue for more information on the encoding. 
+
+    Args:
+      encoded_str (string): string holding specifically encoded environment variable.
+
+    Returns:
+      decoded_str (string): string holding decoded string value. 
+    """
+
+    decoded_str = encoded_str.replace( '%25', '%' )
+    decoded_str = decoded_str.replace( '%0A', '\n' )
+    decoded_str = decoded_str.replace( '%0D', '\r' )
+    return decoded_str
+
+def remove_duplicates( in_dep, in_sum ):
+    """
+    Goes through the dependency lists, checks to see if the dependency listed 
+    already has a ticket capturing the work. 
+      - If yes we remove that dependency from the list and move to the next
+      - If no we leave the dependency in the list and move to the next
+
+    Args: 
+      in_dep (list[string], list[datring]): a list containing tuples of dependency updates
+          and update strings
+      in_sum (list[string]): a list containing all ticket summaries
+
+    Returns: 
+      new_li (list[string]): a list containing only elements that exist from
+          in_dep that did not exist in in_sum
+    """
+
+    # holders for returned list and tickets to only hold dependency name
+    new_li = []
+
+    for ele in in_dep:
+        # get the first elemet of the tuple which is the dependency update
+        dependency = ele[0]
+        # add tuple to list
+        new_li.append( ele )
+
+        # check the dependency for a match and get the dependency
+        check_str = re.search( r"^- `(.*)` Update", dependency )
+        check_str = check_str.group(1)
+        # go through summary list
+        for summary in in_sum:
+            if check_str == summary:
+                # if the dependency is in the summary remove the tuple from the
+                # list and go to the next tuple
+                new_li.remove( ele )
+                break
+    return new_li
+
+def refine_updates( in_dep_str, refine_word ):
+    """
+    Used to parse through incoming dependency update text and pull out the
+    two sections that are surrounded by the argument refine_word. The two
+    sections are then concatanated and returned as a str. 
+
+    Args:
+      in_dep_str (string): string in from environment variable
+      refine_word (string): given word that surrounds the section we are trying to extract
+
+    Returns:
+      refined (string): concatinated dependency updates of the defined refine_word sections
+    """
+
+    refined_str = ""
+    refined_str_dev = ""
+
+    # get the first section of refine_word
+    start = re.search( refine_word, in_dep_str)
+    end = re.search(refine_word + ".:", in_dep_str)
+    # if either are none there is an error or the section doesnt exist
+    if not (start is None or end is None):
```

Comment:
Didn't realize you could use `is` for this. I always used `==`. 👍

Reply:
when using == and None it is testing types, is checks for an exact match of values. (Pylint complains if you use == and None)